src/core/core.h

```diff
@@ -180,14 +192,18 @@ typedef struct _msg_hdr_t {
 	// TODO: non serve davvero, togliere
 	int   			type;
 	unsigned long long	rendezvous_mark;	/// Unique identifier of the message, used for rendez-vous event
-	int			alloc_tid;
+	unsigned int		alloc_tid;
```

Comment:
```suggestion
	unsigned int		alloc_tid;
```

Do we need to send it both in header and in actual msg?

Reply:
We use this information to free a buffer of a positive message once it is annihilated, so it should be kept. The messages should be anyhow rearranged significantly. So far so good for me.